File: typescript/bob/bob.ts
class Bob {
  hey(originalConversation: string): string {
    const conversation = originalConversation.trim();
    const isQuestion = conversation.endsWith("?");
    const hasLetters = conversation.toLowerCase().match(/[a-z]/) != null;
    const isShouted = hasLetters && conversation.toUpperCase() == conversation;
    const isEmpty = conversation.length === 0;

    if (isEmpty) {
      return "Fine. Be that way!";
    }

    if (isShouted && isQuestion) {
      return "Calm down, I know what I\'m doing!";
    }

    if (isShouted) {
      return "Whoa, chill out!";
    }

    if (isQuestion) {
      return "Sure.";
    }

    return "Whatever.";
  }
}

export default Bob;

File: kotlin/bob/src/main/kotlin/Bob.kt
object Bob {
    fun hey(input: String): String {
        val cleanInput = input.trim()
        val isShouted = cleanInput.uppercase() == cleanInput
        val hasLetters = cleanInput.any { it -> it.isLetter() }
        val isQuestion = cleanInput.endsWith("?")
        val isStatement = cleanInput.endsWith("!")

        return when {
            cleanInput.isEmpty() -> "Fine. Be that way!"
            isQuestion && hasLetters && isShouted -> "Calm down, I know what I'm doing!"
            isQuestion -> "Sure."
            isShouted && (isStatement || hasLetters) -> "Whoa, chill out!"
            else -> "Whatever."
        }
    }
}
